spec/lib/rbac/roles_spec.rb

```diff
@@ -0,0 +1,61 @@
+describe RBAC::Roles do
+  let(:api_instance) { instance_double(RBACApiClient::RoleApi) }
+  let(:catalog_admin) do
+    {
+      :data => [
+        {
+          :name        => "Catalog Administrator",
+          :description => "A catalog administrator roles grants read, write and order permissions"
+        }
+      ],
+      :meta => {
+        :count => 1
+      }
+    }
+  end
+  let(:opts) { {:name => 'Catalog Administrator', :scope => 'principal'} }
+  let(:response_headers) { {"Content-Type" => 'application/json'} }
+
+  around do |example|
+    with_modified_env(:RBAC_URL => "http://localhost") do
+      ManageIQ::API::Common::Request.with_request(default_request) { example.call }
+    end
+  end
+
+  before do
+    allow(api_instance).to receive(:list_roles).with(opts).and_return(catalog_admin.as_json)
+  end
+
+  describe "#self.assigned_role?" do
+    let(:assigned_role) { described_class.assigned_role?("Catalog Administrator") }
+
+    context "when the role exists" do
+      before do
+        stub_request(:get, "http://localhost/api/rbac/v1/roles/?limit=10&name=Catalog%20Administrator&offset=0&scope=principal")
+          .to_return(:status  => 200,
+                     :body    => catalog_admin.to_json,
+                     :headers => response_headers)
+      end
+
+      it "returns true for assigned_role" do
+        expect(assigned_role).to be_truthy
+      end
+    end
+
+    context "when the role does not exist" do
+      before do
+        stub_request(:get, "http://localhost/api/rbac/v1/roles/?limit=10&name=Catalog%20Administrator&offset=0&scope=principal")
+          .to_return(:status  => 200,
+                     :body    => catalog_admin.tap do |resp|
+                                   resp[:data] = []
+                                   resp[:meta][:count] = 0
+                                 end.to_json,
+                     :headers => response_headers)
```

Comment:
I think it would be better to just move this `before` up a level, and then modify the body to simply return `catalog_admin.to_json` like on line 36, but then have `let`s for the data and meta count. So it would look more like this:
```ruby
let(:catalog_admin) do
  {:data => data, :meta => {:count => count}}
end

before do
  # stub request
end

context "when the role exists" do
  let(:data) do
    [{
      :name        => "Catalog Administrator",
      :description => "A catalog administrator roles grants read, write and order permissions"
    }]
  end
  let(:count) { 1 }

  # spec
end

context "when the role does not exist" do
  let(:data) { [] }
  let(:count) { 0 }

  # spec
end
```

Reply:
Oh yeah that would looks great, since we're re-using the stuff between the two cases and the only thing that really changes is `data`/`count`. Updated.